### PR TITLE
Fix recursion breaking debugger threads display.

### LINKED ISSUE #4141: Debugger 'threads' menu can't display the same block twice in the same thread
When doing recursion, the same block can be in the stack more than once. This was not accounted for, so when this happens the duplicate blocks do not display.

image

### PR DESCRIPTION
**Pull request by Tacodude7729**
Resolves #4141

### CODE REVIEW

File: addons/debugger/threads.js [hunk 1]
```diff
@@ -148,14 +159,13 @@ export default async function createThreadsTab({ debug, addon, console, msg }) {
       const topBlock = thread.target.blocks.getBlock(thread.topBlock);
       const result = [cacheInfo.headerItem];
       if (topBlock) {
-        concatInPlace(result, createBlockInfo(topBlock, null));
+        concatInPlace(result, createBlockInfo(topBlock, -1));
```
Comment (GarboMuffin) [inline addons/debugger/threads.js:162]:
```suggestion
        concatInPlace(result, createBlockInfo(topBlock, 0));
```

Fixes highlight when top block of script is a breakpoint block

File: addons/debugger/threads.js [hunk 2]
```diff
@@ -119,21 +119,32 @@ export default async function createThreadsTab({ debug, addon, console, msg }) {
...
 
+        const stackFrame = thread.stackFrames[stackFrameIdx];
+
         if (!cacheInfo.blockCache.has(block)) {
-          cacheInfo.blockCache.set(block, {
+          cacheInfo.blockCache.set(block, {});
+        }
+
+        const blockInfoMap = cacheInfo.blockCache.get(block);
+        var blockInfo = blockInfoMap[stackFrameIdx];
```
Comment (GarboMuffin) [inline addons/debugger/threads.js:133]:
```suggestion
        let blockInfo = blockInfoMap[stackFrameIdx];
```
nitpick

**Review by GarboMuffin — CHANGES_REQUESTED** [review]

**Comment by easrng** [comment]
#4142 is a blocker for this

**Comment by RedGuy12** [comment]
> #4142 is a blocker for this

@easrng this is #4142, and this has already been merged.

**Pull request merged**